File: benchmarks/src/lua/string_pipeline.lua
local seed=12345;local line=""
for j=1,40 do seed=(seed*16807)%2147483647;local fn=(seed%1000)+100;line=line.."f"..fn;if j<40 then line=line.."," end end
local R=10000;local sp,si,ss=0,0,0;local start=os.clock()
for r=1,R do local parts={};for part in line:gmatch("[^,]+") do parts[#parts+1]=part end
 for k=1,#parts do sp=sp+#parts;si=si+(parts[k]:find("f")or 1)-1;local p=parts[k];if #p>=4 then ss=ss+#p:sub(2,3) end end
end
local acc=((sp%1000003)+(si%1000003)+(ss%1000003))%1000003;local finish=os.clock()
print("Result: "..acc);print(string.format("Time: %.0fms",(finish-start)*1000))

File: benchmarks/src/nodejs/string_pipeline.js
let seed = 12345;
let line = "";
for (let j = 0; j < 40; j++) { seed = (seed * 16807) % 2147483647; let field_num = (seed % 1000) + 100; line += "f" + field_num; if (j < 39) line += ","; }
const R = 10000;
let sum_plen = 0, sum_idx = 0, sum_sub = 0;
const start = Date.now();
for (let r = 0; r < R; r++) {
    const parts = line.split(",");
    for (let k = 0; k < parts.length; k++) {
        sum_plen += parts.length;
        sum_idx += parts[k].indexOf("f");
        const p = parts[k];
        if (p.length >= 4) { sum_sub += p.substring(1, 3).length; }
    }
}
const acc = ((sum_plen % 1000003) + (sum_idx % 1000003) + (sum_sub % 1000003)) % 1000003;
const end = Date.now();
console.log("Result: " + acc);
console.log("Time: " + Math.floor(end - start) + "ms");
